### PR TITLE
chore: standardize npm script names

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Refers https://github.com/eslint/eslint/issues/14827

#### What changes did you make? (Give an overview)

This updates the names of the scripts in package.json to be consistent with the [new standard](https://eslint.org/docs/latest/contribute/package-json-conventions).

#### Is there anything you'd like reviewers to focus on?
We may need to update the `release` related scripts on Jenkins before merging this PR.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,7 +41,5 @@ jobs:
         node-version: ${{ matrix.node }}
     - name: Install dependencies
       run: npm install
-    - name: Build commonjs
-      run: npm run build
```
Comment (snitin315) [inline .github/workflows/ci.yml:45]: build happens via the `pretest` script.

**Review by mdjermanovic — COMMENTED** [review]
Can you also update README (`npm test` will no longer include linting):

https://github.com/eslint/espree/blob/086555ed960bd9cdbe9818cf962eaa1f5f32b737/README.md?plain=1#L186

File: package.json [hunk 2]
```diff
@@ -67,22 +67,21 @@
...
+    "release:generate:latest": "eslint-generate-release",
+    "release:generate:alpha": "eslint-generate-prerelease alpha",
+    "release:generate:beta": "eslint-generate-prerelease beta",
+    "release:generate:rc": "eslint-generate-prerelease rc",
+    "release:publish": "eslint-publish-release",
+    "sync:docs": "node sync-docs.js",
+    "test": "npm-run-all -s test:*",
+    "test:esm": "c8 mocha --color --reporter progress --timeout 30000 'tests/lib/**/*.js'",
+    "test:cjs": "mocha --color --reporter progress --timeout 30000 tests/lib/commonjs.cjs",
+    "update:version": "node tools/update-version.js"
```
Comment (mdjermanovic) [inline package.json:85]: I think this should be `build:update:version`, or `build:update-version`.

File: package.json [hunk 3]
```diff
@@ -67,22 +67,21 @@
...
-    "generate-alpharelease": "eslint-generate-prerelease alpha",
-    "generate-betarelease": "eslint-generate-prerelease beta",
-    "generate-rcrelease": "eslint-generate-prerelease rc",
-    "publish-release": "eslint-publish-release"
+    "release:generate:latest": "eslint-generate-release",
+    "release:generate:alpha": "eslint-generate-prerelease alpha",
+    "release:generate:beta": "eslint-generate-prerelease beta",
+    "release:generate:rc": "eslint-generate-prerelease rc",
+    "release:publish": "eslint-publish-release",
+    "sync:docs": "node sync-docs.js",
```
Comment (mdjermanovic) [inline package.json:81]: I think this should be `build:docs`

**Review by nzakas — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (snitin315) [reply package.json]: Updated.

File: package.json [hunk 4]
```diff
@@ -67,22 +67,21 @@
...
-    "unit": "npm-run-all -s unit:*",
-    "unit:esm": "c8 mocha --color --reporter progress --timeout 30000 'tests/lib/**/*.js'",
-    "unit:cjs": "mocha --color --reporter progress --timeout 30000 tests/lib/commonjs.cjs",
-    "test": "npm-run-all -p unit lint",
-    "lint": "eslint .  --report-unused-disable-directives",
-    "fixlint": "npm run lint -- --fix",
     "build": "rollup -c rollup.config.js",
     "build:debug": "npm run build -- -m",
-    "update-version": "node tools/update-version.js",
+    "build:docs": "node sync-docs.js",
```
Comment (mdjermanovic) [inline package.json:72]:
```suggestion
    "build:docs": "node tools/sync-docs.js",
```

While here, we could fix this script.

Comment (snitin315) [reply package.json]: Good catch.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for @nzakas to verify.

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Comment by mdjermanovic** [comment]
> We may need to update the `release` related scripts on Jenkins before merging this PR.

Updated.

**Pull request merged**